lib/systems/parse.nix

```diff
@@ -444,7 +444,7 @@ rec {
       else throw "Target specification with 1 components is ambiguous";
     "2" = # We only do 2-part hacks for things Nix already supports
       if elemAt l 1 == "cygwin"
-        then { cpu = elemAt l 0;                      kernel = "windows";  abi = "cygnus";   }
+        then mkSkeletonFromList [ (elemAt l 0) "pc" "cygwin" ]
```

Comment:
This change is purely a matter of style, not behavior, right?

(I kinda wonder if we can just drop this case, as the older cygwin stuff is bit-rotted for years at this point.)

Reply:
I'm in favour of getting rid of all the older Cygwin support, since it just doesn't work anymore (I tried).

Reply:
Yeah it's well over a decade old

Reply:
I had a go at removing the hack altogether: https://github.com/puffnfresh/nixpkgs/commit/16e9e8bbb798604a10e8e29706b0a94146b04ba1

It feels a little bit wrong to add Cygwin as a "kernel" but it does seem to work. Let me know what you think.